lib/ruby_lsp/utils.rb

```diff
@@ -6,7 +6,7 @@ module RubyLsp
   VOID = T.let(Object.new.freeze, Object)
 
   # This freeze is not redundant since the interpolated string is mutable
-  WORKSPACE_URI = T.let(URI("file://#{Dir.pwd}".freeze), URI::Generic)
+  WORKSPACE_URI = T.let(URI(URI::Parser.new.escape("file://#{Dir.pwd}".freeze)), URI::Generic)
```

Comment:
I have a question regarding this one: does the editor send spaces encoded or unencoded? That is, do we receive `file:///something otherthing/blah.rb` or `file:///something%20otherthing/blah.rb`?

And does `URI()` not handle does already? If I'm not mistaken `URI()` invokes the parser, so I'm surprised it doesn't handle the spaces.

Reply:
The URI gem does not have a better answer to this. I created https://github.com/ruby/uri/issues/87 over there.